core/src/network_manager/manager.rs

```diff
@@ -376,23 +428,35 @@ impl NetworkManagerExecutor {
                 let mpc_net = match local_role {
                     ConnectionRole::Dialer => {
                         // Retrieve known dialable addresses for the peer from the network behavior
-                        let all_peer_addrs = self.swarm.behaviour_mut().addresses_of_peer(&peer_id);
-                        let peer_multiaddr = all_peer_addrs.get(0).ok_or_else(|| {
-                            NetworkManagerError::Network(ERR_NO_KNOWN_ADDR.to_string())
-                        })?;
-                        let peer_addr = multiaddr_to_socketaddr(peer_multiaddr.clone(), peer_port)
+                        let all_peer_addrs = self
+                            .swarm
+                            .behaviour_mut()
+                            .handle_pending_outbound_connection(
+                                ConnectionId::new_unchecked(0),
+                                Some(peer_id.0),
+                                &[],
+                                Endpoint::Dialer,
+                            )
+                            .map_err(|_| {
+                                NetworkManagerError::Network(ERR_NO_KNOWN_ADDR.to_string())
+                            })?;
+
+                        // Map each resolved address into a SocketAddr and remove local addresses
+                        let peer_addr = all_peer_addrs
+                            .into_iter()
+                            .find(|addr| !is_local_addr(addr))
+                            .map(|addr| multiaddr_to_socketaddr(&addr, peer_port))
+                            .unwrap_or(None)
                             .ok_or_else(|| {
-                                NetworkManagerError::SerializeDeserialize(
-                                    ERR_PARSING_ADDR.to_string(),
-                                )
```

Comment:
Can we still get serialization/deserialization errors in `multiaddr_to_socketaddr`?
It's nice to have specific errors and idk if `ERR_NO_KNOWN_ADDR` is exactly fitting here.

Regardless though this is pretty nitpicky so feel free to dismiss

Reply:
Yeah good suggestion, fixed